File: sample-conference-videochat/sample-conference-videochat/Components/ChatScreen/Menu Action/MenuAction.m
//
//  MenuAction.m
//  sample-conference-videochat
//
//  Created by Injoit on 08.10.2020.
//  Copyright © 2020 Quickblox. All rights reserved.
//

#import "MenuAction.h"

@implementation MenuAction

- (instancetype)initWithTitle:(NSString *)title action:(ChatAction)action handler:(MenuActionHandler _Nullable)handler {
    self = [super init];
    if (self) {
        self.title = title;
        self.action = action;
        self.handler = handler;
    }
    return self;
}

@end

File: sample-conference-videochat-swift/sample-conference-videochat-swift/Components/ChatScreen/ActionsMenuViewController/MenuAction.swift
//
//  MenuAction.swift
//  sample-conference-videochat-swift
//
//  Created by Injoit on 15.08.2020.
//  Copyright © 2020 QuickBlox. All rights reserved.
//

import Foundation

typealias SelectedAction = ((_ action: ChatAction) -> Void)

class MenuAction {
    //MARK: - Properties
    var successHandler: SelectedAction?
    var title: String
    var action: ChatAction
    
    //MARK: - Life Cycle
    init (title: String,
          action: ChatAction,
          handler: @escaping SelectedAction) {
        self.title = title
        self.action = action
        self.successHandler = handler
    }
}
